### ISSUE TITLE
Make the tool work with "cellranger vdj" bam outputs

### ISSUE THREAD

**Issue by sralchemab** [10XGenomics/bamtofastq#108]
It would be great if the tool could produce a FastQ file from the VDJ pipeline BAM files (e.g. `all_contig.bam`), including the usage of the `--bx-list` option as well.

Thanks in advance.

**Comment by apredeus** [comment]
I agree, it would be great to have a tool like that. Often users submit BAM files to GEO/SRA, and their decoding is, to put it mildly, often _inconsistent_. E.g. in this project, there are 14 10x VDJ experiments, 12 of which were decoded as paired-end reads, and 2 as single-end read for some reason. 

Are there any tools that decode VDJ BAMs to paired-end reads correctly? Also, why is there a cap on how many reads per barcode are outputted to the `all_contig.bam` file? It's so great that a regular 10x BAM file contains all of the reads, makes uploading a BAM or original fastq files truly interchangeable. However it causes lots of confusion, because most of the raw read upload systems allow you to upload a BAM instead of fastq files.
